### PR TITLE
[AI-481] Rio-Tiler Clipped Tiles

### PR DESCRIPTION
**Pull request by jwiem**
## What?

This PR adds the ability to clip tiles to a GeoJSON boundary with rio-tiler. Rio-tiler can take a GeoJSON feature or more specifically, just a `geometry as input. From the tests, riotiler will tile both a regular image and any type of band expression. 

image

### CODE REVIEW

**Review by tnigon — APPROVED** [review]

File: rio_tiler/io/cogeo.py [hunk 1]
```diff
@@ -346,6 +348,24 @@ def tile(
...
+                bounds_crs=None,
+                height=tilesize,
+                width=tilesize,
+                max_size=None,
+                indexes=indexes,
+                expression=expression,
+                vrt_options=vrt_options,
+                **kwargs,
+            )
+
```
Comment (DanSchoppe) [inline rio_tiler/io/cogeo.py]: I don't like that `aio` forces you to repeat the 10-line return statement verbatim with the only change being for `vrt_options`. How about:
```suggestion
        vrt_options = None
        if aoi is not None:
            cutline = create_cutline(self.dataset, aoi, geometry_crs="epsg:4326")
            vrt_options = kwargs.pop("vrt_options", {})
            vrt_options.update({"cutline": cutline})

```
Then just passing `vrt_options=vrt_options` in the single return

Comment (DanSchoppe) [reply rio_tiler/io/cogeo.py]: Or better yet, assuming there's no reason we explicitly don't want kwargs vrt_options passed through:
```
vrt_options = kwargs.pop("vrt_options", {})
if aoi is not None:
    cutline = ...
    vrt_options.update(...)

return self.part(
    ...
    vrt_options=vrt_options
    ...
)
```

Comment (jwiem) [reply rio_tiler/io/cogeo.py]: Thanks for that. That's exactly what I was looking for 😄  e670c1e05f249493adc4144f291364fa64f14cc0

File: docs/src/mosaic.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ Inputs:
 - **chunk_size**: optional, control the number of assets to process per loop.
 - **threads**: optional, number of threads to use in each loop.
 - **allowed_exceptions**: optional, allow some exceptions to be ignored.
-- **\*kwargs**: tiler specific keyword arguments.
+- **\*\*kwargs**: tiler specific keyword arguments.
```
Comment (DanSchoppe) [inline docs/src/mosaic.md]: Was this change intentional?

**Review by DanSchoppe — APPROVED** [review]

File: docs/src/mosaic.md (hunk 2 above)
Comment (jwiem) [reply docs/src/mosaic.md]: No, must've been pre-commit.

Comment (DanSchoppe) [inline docs/src/mosaic.md]: https://github.com/SenteraLLC/rio-tiler/pull/1/files#r878477063

```suggestion
- **\*kwargs**: tiler specific keyword arguments.
```
